src/System.Reflection.Metadata/src/System/Reflection/Metadata/TypeSystem/AssemblyReference.cs

```diff
@@ -122,6 +122,11 @@ public BlobHandle HashValue
             }
         }
 
+        public AssemblyName GetAssemblyName()
+        {
+            return _reader.GetAssemblyName();
+        }
```

Comment:
Actually, you probably still want a shared static internal helper that takes String and blob handles for name, culture, token, etc. and produces an AssemblyName.

The key point is that each AssemblyReference has a distinct AssemblyName and they're all distinct from the AssemblyName of the AssemblyDefinition.